polaris-service/src/main/java/org/apache/polaris/service/context/DefaultContextResolver.java

```diff
@@ -92,14 +92,22 @@ public RealmContext resolveRealmContext(
 
     if (!parsedProperties.containsKey(REALM_PROPERTY_KEY)) {
       LOGGER.warn(
-          "Failed to parse {} from headers; using {}",
-          REALM_PROPERTY_KEY,
-          REALM_PROPERTY_DEFAULT_VALUE);
-      parsedProperties.put(REALM_PROPERTY_KEY, REALM_PROPERTY_DEFAULT_VALUE);
+          "Failed to parse {} from headers; using {}", REALM_PROPERTY_KEY, getDefaultRealm());
+      parsedProperties.put(REALM_PROPERTY_KEY, getDefaultRealm());
     }
     return () -> parsedProperties.get(REALM_PROPERTY_KEY);
   }
 
+  @Override
+  public void setDefaultRealm(String defaultRealm) {
+    this.defaultRealm = defaultRealm;
+  }
+
+  @Override
+  public String getDefaultRealm() {
+    return this.defaultRealm;
+  }
```

Comment:
Can we remove this method if it is only used within this class?

Reply:
I don't think we should keep the setter but not the getter